src/VSTS-Bot.Api/Dialogs/SubscriptionsDialog.cs

```diff
@@ -0,0 +1,289 @@
+            result.ThrowIfNull(nameof(result));
+
+            var activity = await result;
+            var text = (activity.RemoveRecipientMention() ?? string.Empty).ToLowerInvariant();
+            var reply = context.MakeMessage();
+
+            var matchSubscribe = Regex.Match(text, CommandMatchSubscribe);
+            var matchUnsubscribe = Regex.Match(text, CommandMatchUnsubscribe);
+
+            if (matchSubscribe.Success || matchUnsubscribe.Success)
+            {
+                var typing = context.MakeMessage();
+                typing.Type = ActivityTypes.Typing;
+                await context.PostAsync(typing);
+            }
+
+            if (matchSubscribe.Success)
+            {
+                var subscriptionType = (SubscriptionType)Enum.Parse(typeof(SubscriptionType), matchSubscribe.Groups[1].Value, true);
+                var querySpec = new SqlQuerySpec
+                {
+                    QueryText = "SELECT * FROM subscriptions s WHERE s.channelId = @channelId AND s.userId = @userId AND s.subscriptionType = @subscriptionType",
+                    Parameters = new SqlParameterCollection
+                    {
+                        new SqlParameter("@channelId", activity.ChannelId),
+                        new SqlParameter("@userId", activity.From.Id),
+                        new SqlParameter("@subscriptionType", subscriptionType)
+                    }
+                };
+
+                var subscription = this.documentClient
+                    .CreateDocumentQuery<Subscription>(UriFactory.CreateDocumentCollectionUri("botdb", "subscriptioncollection"), querySpec)
+                    .ToList()
+                    .FirstOrDefault();
+
+                if (subscription != null)
+                {
+                    context.Done(reply);
+                    return;
+                }
+
+                var teamProjects = await this.VstsService.GetProjects(this.Account, this.Profile.Token);
+                var teamProject = teamProjects.FirstOrDefault(tp => tp.Name.Equals(this.TeamProject, StringComparison.OrdinalIgnoreCase));
+
+                if (teamProject == null)
+                {
+                    context.Done(reply);
+                    return;
+                }
+
+                subscription = new Subscription
+                {
+                    BotId = activity.Recipient.Id,
+                    BotName = activity.Recipient.Name,
+                    ChannelId = activity.ChannelId,
+                    IsActive = true,
+                    ProfileId = this.Profile.Id,
+                    RecipientId = activity.From.Id,
+                    RecipientName = activity.From.Name,
+                    ServiceUri = new Uri(activity.ServiceUrl),
+                    SubscriptionType = subscriptionType,
+                    UserId = activity.From.Id
+                };
+
+                var strategy = this.strategies.First(s => s.CanGetSubscription(subscriptionType));
```

Comment:
Of loading to a strategy, which contains the logic what details needs to be in the subscription.